src/test/java/com/wikia/webdriver/testcases/adstests/TestUrlBuilder.java

```diff
@@ -259,121 +263,109 @@
-                String.format("http://%s.sandbox-s1.wikia.com", Configuration.getWikiName()));
-        Assertion.assertEquals(new UrlBuilder("prod", false).getUrlForWiki(),
-                String.format("http://%s.wikia.com", Configuration.getWikiName()));
-        Assertion.assertEquals(new UrlBuilder("preview", true).getUrlForWiki(),
-                String.format("https://%s.preview.wikia.com", Configuration.getWikiName()));
-        Assertion.assertEquals(new UrlBuilder("sandbox-s1", true).getUrlForWiki(),
-                String.format("https://%s.sandbox-s1.wikia.com", Configuration.getWikiName()));
-        Assertion.assertEquals(new UrlBuilder("prod", true).getUrlForWiki(),
-                String.format("https://%s.wikia.com", Configuration.getWikiName()));
-        Assertion.assertEquals(new UrlBuilder("preview", true, true).getUrlForWiki(),
-                String.format("https://%s.preview.wikia.com", Configuration.getWikiName()));
-        Assertion.assertEquals(new UrlBuilder("sandbox-s1", true, true).getUrlForWiki(),
-                String.format("https://%s.sandbox-s1.wikia.com", Configuration.getWikiName()));
-        Assertion.assertEquals(new UrlBuilder("prod", true, true).getUrlForWiki(),
-                String.format("https://%s.wikia.com", Configuration.getWikiName()));
+    private Constructor<UrlBuilder> getUrlBuilderConstructor() throws NoSuchMethodException {
+        Constructor<UrlBuilder> c = UrlBuilder.class
+                .getDeclaredConstructor(String.class, String.class, Boolean.class, Boolean.class, String.class);
+        c.setAccessible(true);
+        return c;
     }
 
     @Test(groups = {"TestUrlBuilder", "unitTests"})
-    public void getUrlForPathTests() {
-
-        Assertion.assertEquals(new UrlBuilder("preview", false).getUrlForPath("/wiki/Special:Version"),
-                String.format("http://%s.preview.wikia.com/wiki/Special:Version", Configuration.getWikiName()));
+    public void getUrlForPathTests() throws Exception {
+        Constructor<UrlBuilder> c = getUrlBuilderConstructor();
+        
+        Assert.assertEquals(c.newInstance("mediawiki119","preview", false, false, DEFAULT_LANGUAGE).getUrlForPath("/wiki/Special:Version"),
+               "http://mediawiki119.preview.wikia.com/wiki/Special:Version");
 
         String cb = "cb=1111";
         Configuration.setTestValue("qs", cb);
 
-        Assertion.assertEquals(new UrlBuilder("prod", false).getUrlForPath("wowwiki", "Portal:Main"),
-                "http://wowwiki.wikia.com/Portal:Main?" + cb);
+        Assert.assertEquals(c.newInstance("mediawiki119","prod", false, false, DEFAULT_LANGUAGE).getUrlForPath("Portal:Main"),
+                "http://mediawiki119.wikia.com/Portal:Main?" + cb);
     }
 
     @Test(groups = {"TestUrlBuilder", "unitTests"})
-    public void getUrlForPathWithLanguageTests() {
-
-        Configuration.setTestValue("language", "szl");
+    public void getUrlForPathWithLanguageTests() throws Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS00112)